src/pipes.cpp

```diff
@@ -0,0 +1,66 @@
+#include "pipes.h"
+
+const int Pipes::NUM_PIPES = 4;
+const float Pipes::PIPE_SCALE = 0.4;
+const int Pipes::PIPE_OFFSET_RANGE = 300;
+const float Pipes::SPEED_PER_SECOND = 500;
+
+Pipes::Pipes(int screenWidth, int screenHeight, int frameRate)
+{
+    bool res = pipeTexture.loadFromFile("./lib/pipe.png");
+    if(!res) throw("Cannot load pipe\n");
+
+    pipeSprites = new sf::Sprite[NUM_PIPES];
+    if (pipeSprites == nullptr) throw("Pipe array could not be created\n");
+
+    for(int i = 0 ; i < NUM_PIPES ; i++)
+    {
+        this->pipeSprites[i].setTexture(pipeTexture, true);
```

Comment:
I hate that you use `this` everywhere. Please find and replace all instances of `this->` with nothing.. It just makes the code wordier. There is no use case for explicit `this` in all of your code